### PR TITLE
Adding singletons for servicebusclient and sender.

### PR DESCRIPTION
**Pull request by xnetsrak**
## References

* #326

### CODE REVIEW

**Review by defectiveAi — DISMISSED** [review]

File: source/Energinet.DataHub.MessageHub.Core.Tests/Factories/AzureServiceBusFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+using Energinet.DataHub.MessageHub.Core.Factories;
+using Energinet.DataHub.MessageHub.Core.Tests.Peek;
+using Energinet.DataHub.MessageHub.Model.Protobuf;
+using Google.Protobuf;
+using Moq;
+using Xunit;
+
+namespace Energinet.DataHub.MessageHub.Core.Tests.Factories
+{
+    public sealed class AzureServiceBusFactoryTests
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Core.Tests/Factories/AzureServiceBusFactoryTests.cs:27]: Attribute `[UnitTest]` missing

**Review by defectiveAi — DISMISSED** [review]

File: source/Energinet.DataHub.MessageHub.Client/source/Energinet.DataHub.MessageHub.Client/DataAvailable/DataAvailableNotificationSender.cs [hunk 2]
```diff
@@ -57,16 +54,7 @@ public async Task SendAsync(DataAvailableNotificationDto dataAvailableNotificati
...
-        }
-
-        public async ValueTask DisposeAsync()
-        {
-            if (_serviceBusClient != null)
-            {
-                await _serviceBusClient.DisposeAsync().ConfigureAwait(false);
-                _serviceBusClient = null;
-            }
+            await sender.PublishMessageAsync<ServiceBusMessage>(message).ConfigureAwait(false);
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Client/source/Energinet.DataHub.MessageHub.Client/DataAvailable/DataAvailableNotificationSender.cs:57]: You could save the `await` here, and just return the `Task`

File: source/Energinet.DataHub.MessageHub.Client/source/Energinet.DataHub.MessageHub.Client/Factories/AzureServiceBusFactory.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            if (serviceBusClientFactory == null) throw new ArgumentNullException(nameof(serviceBusClientFactory));
+
+            _client = serviceBusClientFactory.Create();
+        }
+
+        public ISenderMessageBus GetSenderClient(string queueOrTopicName)
+        {
+            var key = $"{queueOrTopicName}";
+
+            var sender = _senders.GetOrAdd(key, k => _client.CreateSender(queueOrTopicName));
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Client/source/Energinet.DataHub.MessageHub.Client/Factories/AzureServiceBusFactory.cs:39]: `_ => ...`

File: source/Energinet.DataHub.MessageHub.Client/source/Energinet.DataHub.MessageHub.Client/Factories/AzureSessionReceiverServiceBus.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+
+namespace Energinet.DataHub.MessageHub.Client.Factories
+{
+    public sealed class AzureSessionReceiverServiceBus : IReceiverMessageBus
+    {
+        private readonly ServiceBusSessionReceiver _serviceBusSessionReceiver;
+
+        internal AzureSessionReceiverServiceBus(ServiceBusSessionReceiver serviceBusSessionReceiver)
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Client/source/Energinet.DataHub.MessageHub.Client/Factories/AzureSessionReceiverServiceBus.cs:25]: Can be made `private`

File: source/Energinet.DataHub.MessageHub.Core/source/Energinet.DataHub.MessageHub.Core/Dequeue/DequeueNotificationSender.cs [hunk 5]
```diff
@@ -49,16 +47,7 @@ public async Task SendAsync(DequeueNotificationDto dequeueNotificationDto, Domai
...
-        }
-
-        public async ValueTask DisposeAsync()
-        {
-            if (_serviceBusClient != null)
-            {
-                await _serviceBusClient.DisposeAsync().ConfigureAwait(false);
-                _serviceBusClient = null;
-            }
+            await serviceBusSender.PublishMessageAsync<ServiceBusMessage>(dequeueMessage).ConfigureAwait(false);
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Core/source/Energinet.DataHub.MessageHub.Core/Dequeue/DequeueNotificationSender.cs:50]: Return `Task` without awaiting it

File: source/Energinet.DataHub.MessageHub.Core/source/Energinet.DataHub.MessageHub.Core/Factories/AzureSenderServiceBus.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+
+namespace Energinet.DataHub.MessageHub.Core.Factories
+{
+    public sealed class AzureSenderServiceBus : ISenderMessageBus
+    {
+        private readonly ServiceBusSender _serviceBusSender;
+
+        internal AzureSenderServiceBus(ServiceBusSender serviceBusSender)
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Core/source/Energinet.DataHub.MessageHub.Core/Factories/AzureSenderServiceBus.cs:24]: Can be made `private`

File: source/Energinet.DataHub.MessageHub.Core/source/Energinet.DataHub.MessageHub.Core/Factories/AzureSessionReceiverServiceBus.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+
+namespace Energinet.DataHub.MessageHub.Core.Factories
+{
+    public sealed class AzureSessionReceiverServiceBus : IReceiverMessageBus
+    {
+        private readonly ServiceBusSessionReceiver _serviceBusSessionReceiver;
+
+        internal AzureSessionReceiverServiceBus(ServiceBusSessionReceiver serviceBusSessionReceiver)
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MessageHub.Core/source/Energinet.DataHub.MessageHub.Core/Factories/AzureSessionReceiverServiceBus.cs:25]: Can be made `private`

**Review by michaelmajgaard — APPROVED** [review]

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**